luminoso_api/client.py

```diff
@@ -487,14 +489,34 @@ def get_token_filename():
                         '.luminoso',
                         'tokens.json')
 
-def get_root_url(url):
+def get_root_url(url, warn=True):
     """
-    If we have to guess a root URL, assume it contains the scheme,
-    hostname, and one path component, as in "https://api.luminoso.com/v4".
+    Guessing the root URL is slightly complicated, because the API may be at
+    /v4 or at /api/v4, depending on whether you're accessing the API through
+    its old URL or its new one.
     """
-    # make sure it's a complete URL, not a relative one
-    assert ':' in url
-    return '/'.join(url.split('/')[:4])
+    parsed_url = urlparse(url)
+
+    # Make sure it's a complete URL, not a relative one.
+    if not parsed_url.scheme:
+        raise ValueError('Please supply a full URL, beginning with http:// '
+                         'or https:// .')
+
+    # If the path starts with v4, use that for the base URL, but issue a
+    # warning about the URL moving.
+    if parsed_url.path.startswith('/v4'):
+        if warn:
+            logger.warning('While the old URL will continue to work, please '
+                           'be aware that the Luminoso API has moved to %s',
+                           URL_BASE)
```

Comment:
This warning *only* applies to the deployed API (and maybe staging?), right?  Like, if I'm reusing lines from my ipython history and try to connect to "http://my-computer:5000/v4", it will print this warning and then try to use a URL that won't work?